tests/phpunit/CRM/Contribute/BAO/ContributionTest.php

```diff
@@ -664,9 +664,6 @@ public function checkItemValues($contribution) {
       $this->assertEquals(150.00, $dao->total_amount, 'Mismatch of total amount paid.');
       $this->assertEquals($dao->amount, array_pop($amounts), 'Mismatch of amount proportionally assigned to financial item');
     }
-
-    Contact::delete($this->_contactId);
-    Event::delete($this->_eventId);
```

Comment:
Good spotting. Since this class was written Tim introduced a second form of cleanup which is rollback - ie. at the end of the test all transactions within it are rolled back. I 

https://github.com/civicrm/civicrm-core/blob/master/tests/phpunit/CiviTest/CiviUnitTestCase.php#L44 

I thought this was kicking in here - but I can't find evidence of it so it must be just 'getting away with it'.

I'll try adding that in though.... - which will mean we need much less cleanup in this class if it works

  /**
   * Test setup for every test.
   */
  public function setUp() {
    parent::setUp();
    $this->useTransaction(TRUE);
  }